ibis/common/grounds.py

```diff
@@ -69,17 +69,28 @@ def validate(self, this, arg):
             return self.validator(arg, this=this)
 
 
-@functools.singledispatch
-def _get_hashable_object(value: Any) -> Hashable:
-    return value
+class Immutable(Hashable):
 
+    __slots__ = ()
+
+    def __setattr__(self, name: str, _: Any) -> None:
+        raise TypeError(
+            f"Attribute {name!r} cannot be assigned to immutable instance of "
+            f"type {type(self)}"
+        )
+
+
+# @functools.singledispatch
+# def _get_hashable_object(value: Any) -> Hashable:
+#     return value
 
-@_get_hashable_object.register(pd.DataFrame)
-def _get_hashable_object_dataframe(df: pd.DataFrame) -> Hashable:
-    # XXX: is this an acceptable approach?
-    buf = io.BytesIO()
-    df.to_pickle(buf)
-    return buf.getvalue()
+
+# @_get_hashable_object.register(pd.DataFrame)
+# def _get_hashable_object_dataframe(df: pd.DataFrame) -> Hashable:
```

Comment:
`ibis.common.grounds` shouldn't depend on pandas in any way

Reply:
That's fine. I guess we can move this to some other place then.